### ISSUE TITLE
Mastodon clear cache didn't work during update to v3.3.0

### ISSUE THREAD

**Issue by pixel8383** [tootsuite/mastodon#15463]
I tried to follow the (poor) instructions shown on the v3.3.0 release in order to upgrade from v3.2.1

When I finally tried to clear the cache, I got the following result and what I got is that the Notification/mention history for the users is empty. How can I check if everything is working fine? How can I retrieve that history?

```
mastodon@mastodon-678611:~/live$ RAILS_ENV=production bin/tootctl cache clear
Traceback (most recent call last):
        13: from bin/tootctl:4:in `<main>'
        12: from bin/tootctl:4:in `require_relative'
        11: from /home/mastodon/live/config/boot.rb:9:in `<top (required)>'
        10: from /home/mastodon/live/config/boot.rb:9:in `require'
         9: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap.rb:5:in `<top (required)>'
         8: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap.rb:5:in `require_relative'
         7: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache.rb:74:in `<top (required)>'
         6: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache.rb:74:in `require_relative'
         5: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/store.rb:4:in `<top (required)>'
         4: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/explicit_require.rb:41:in `with_gems'
         3: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/store.rb:4:in `block in <top (required)>'
         2: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/store.rb:4:in `require'
         1: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/msgpack-1.3.3/lib/msgpack.rb:9:in `<top (required)>'
/home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/msgpack-1.3.3/lib/msgpack.rb:9:in `require': cannot load such file -- msgpack/2.7/msgpack (LoadError)
        14: from bin/tootctl:4:in `<main>'
        13: from bin/tootctl:4:in `require_relative'
        12: from /home/mastodon/live/config/boot.rb:9:in `<top (required)>'
        11: from /home/mastodon/live/config/boot.rb:9:in `require'
        10: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap.rb:5:in `<top (required)>'
         9: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap.rb:5:in `require_relative'
         8: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache.rb:74:in `<top (required)>'
         7: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache.rb:74:in `require_relative'
         6: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/store.rb:4:in `<top (required)>'
         5: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/explicit_require.rb:41:in `with_gems'
         4: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/store.rb:4:in `block in <top (required)>'
         3: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/store.rb:4:in `require'
         2: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/msgpack-1.3.3/lib/msgpack.rb:8:in `<top (required)>'
         1: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/msgpack-1.3.3/lib/msgpack.rb:11:in `rescue in <top (required)>'
/home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/msgpack-1.3.3/lib/msgpack.rb:11:in `require': incompatible library version - /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/msgpack-1.3.3/lib/msgpack/msgpack.so (LoadError)
        14: from bin/tootctl:4:in `<main>'
        13: from bin/tootctl:4:in `require_relative'
        12: from /home/mastodon/live/config/boot.rb:9:in `<top (required)>'
        11: from /home/mastodon/live/config/boot.rb:9:in `require'
        10: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap.rb:5:in `<top (required)>'
         9: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap.rb:5:in `require_relative'
         8: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache.rb:74:in `<top (required)>'
         7: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache.rb:74:in `require_relative'
         6: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/store.rb:4:in `<top (required)>'
         5: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/explicit_require.rb:40:in `with_gems'
         4: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/explicit_require.rb:44:in `rescue in with_gems'
         3: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/store.rb:4:in `block in <top (required)>'
         2: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/store.rb:4:in `require'
         1: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/msgpack-1.3.3/lib/msgpack.rb:9:in `<top (required)>'
/home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/msgpack-1.3.3/lib/msgpack.rb:9:in `require': cannot load such file -- msgpack/2.7/msgpack (LoadError)
        15: from bin/tootctl:4:in `<main>'
        14: from bin/tootctl:4:in `require_relative'
        13: from /home/mastodon/live/config/boot.rb:9:in `<top (required)>'
        12: from /home/mastodon/live/config/boot.rb:9:in `require'
        11: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap.rb:5:in `<top (required)>'
        10: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap.rb:5:in `require_relative'
         9: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache.rb:74:in `<top (required)>'
         8: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache.rb:74:in `require_relative'
         7: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/store.rb:4:in `<top (required)>'
         6: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/explicit_require.rb:40:in `with_gems'
         5: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/explicit_require.rb:44:in `rescue in with_gems'
         4: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/store.rb:4:in `block in <top (required)>'
         3: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/store.rb:4:in `require'
         2: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/msgpack-1.3.3/lib/msgpack.rb:8:in `<top (required)>'
         1: from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/msgpack-1.3.3/lib/msgpack.rb:11:in `rescue in <top (required)>'
/home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/msgpack-1.3.3/lib/msgpack.rb:11:in `require': incompatible library version - /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/msgpack-1.3.3/lib/msgpack/msgpack.so (LoadError)
```

**Comment by Gargron** [comment]
> incompatible library version - /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/msgpack-1.3.3/lib/msgpack/msgpack.so (LoadError)

Have you run `bundle install` before running this?

**Comment by pixel8383** [comment]
Yes I did, the actual release shown on the web interface is 3.3.0

**Comment by Gargron** [comment]
I am not familiar with this error and I would not expect the web process to launch if tootctl cannot launch. I can only advise that errors about missing .so files sometimes happen when system libraries are updated without recompiling bundler dependencies, i.e. if you upgrade Ubuntu 18 to 20 and leave everything as is it will start complaining about .so files being wrong. The solution to that is something like `bundle install --force` which recompiles everything. See e.g. #14987

**Comment by pixel8383** [comment]
If I try to execute `bundle install --force` what I get is:

```
mastodon@mastodon-678611:~/live$ bundle install --force
[DEPRECATED] The `--force` option has been renamed to `--redownload`
Fetching gem metadata from https://rubygems.org/............
Installing rake 13.0.1
Installing concurrent-ruby 1.1.7
Installing i18n 1.8.5
Installing minitest 5.14.2
Installing thread_safe 0.3.6
Installing tzinfo 1.2.7
Installing activesupport 5.2.4.4
Installing builder 3.2.4
Installing erubi 1.9.0
Installing mini_portile2 2.4.0
Fetching nokogiri 1.10.10
Installing nokogiri 1.10.10 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/nokogiri-1.10.10/ext/nokogiri
/home/mastodon/.rbenv/versions/2.7.2/bin/ruby -I /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0 -r ./siteconf20201230-224468-1pg33y7.rb
extconf.rb
checking if the C compiler accepts ... yes
Building nokogiri using packaged libraries.
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
        --with-opt-dir
        --without-opt-dir
        --with-opt-include
        --without-opt-include=${opt-dir}/include
        --with-opt-lib
        --without-opt-lib=${opt-dir}/lib
        --with-make-prog
        --without-make-prog
        --srcdir=.
        --curdir
        --ruby=/home/mastodon/.rbenv/versions/2.7.2/bin/$(RUBY_BASE_NAME)
        --help
        --clean
        --use-system-libraries
/home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:168:in `require': incompatible library version -
/home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/openssl-2.2.0/lib/openssl.so (LoadError)
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:168:in `rescue in require'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:156:in `require'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/openssl.rb:13:in `<top (required)>'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/net/https.rb:23:in `<top (required)>'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
        from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/mini_portile2-2.4.0/lib/mini_portile2/mini_portile.rb:3:in `<top (required)>'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
        from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/mini_portile2-2.4.0/lib/mini_portile2.rb:2:in `<top (required)>'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
        from extconf.rb:470:in `<main>'
/home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require': incompatible library version -
/home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/openssl-2.2.0/lib/openssl.so (LoadError)
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/openssl.rb:13:in `<top (required)>'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/net/https.rb:23:in `<top (required)>'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
        from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/mini_portile2-2.4.0/lib/mini_portile2/mini_portile.rb:3:in `<top (required)>'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
        from /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/mini_portile2-2.4.0/lib/mini_portile2.rb:2:in `<top (required)>'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
        from /home/mastodon/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
        from extconf.rb:470:in `<main>'

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /home/mastodon/live/vendor/bundle/ruby/2.7.0/extensions/x86_64-linux/2.7.0/nokogiri-1.10.10/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in /home/mastodon/live/vendor/bundle/ruby/2.7.0/gems/nokogiri-1.10.10 for inspection.
Results logged to /home/mastodon/live/vendor/bundle/ruby/2.7.0/extensions/x86_64-linux/2.7.0/nokogiri-1.10.10/gem_make.out

An error occurred while installing nokogiri (1.10.10), and Bundler cannot continue.
Make sure that `gem install nokogiri -v '1.10.10' --source 'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  health_check was resolved to 4.0.0.pre, which depends on
    rails was resolved to 5.2.4.4, which depends on
      actioncable was resolved to 5.2.4.4, which depends on
        actionpack was resolved to 5.2.4.4, which depends on
          actionview was resolved to 5.2.4.4, which depends on
            rails-dom-testing was resolved to 2.0.3, which depends on
              nokogiri
mastodon@mastodon-678611:~/live$

```

**Comment by jakesullyneytiri** [comment]
first of all you are runing ruby 2.7.0, it requires ruby 2.7.2

So make sure you have that first.

**Comment by pixel8383** [comment]
I updated ruby to 2.7.2, I can’t see ruby 2.7.0 under .rbenv bin folder.

**Comment by jakesullyneytiri** [comment]
> I updated ruby to 2.7.2, I can’t see ruby 2.7.0 under .rbenv bin folder.
> […](#)
> Il giorno 30 dic 2020, alle ore 12:13, jakesullyneytiri ***@***.***> ha scritto: ﻿ first of all you are runing ruby 2.7.0, it requires ruby 2.7.2 So make sure you have that first. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

my reply was placed above yours, you can check above your reply for solution.

**Comment by pixel8383** [comment]
I managed to restore the machine from a snapshot and I started from scratch from v3.2.2, then I did the following:

1. su - mastodon
2. git -C /home/mastodon/.rbenv/plugins/ruby-build pull
3.  rbenv install 2.7.2
4. stop mastodon services
5. rbenv uninstall 2.6.6
6. git fetch --tags
7. git checkout v3.3.0 
8. bundle install

I got stuck in here with:

```
mastodon@mastodon:~/live$ bundle install
Fetching gem metadata from https://rubygems.org/............
Using rake 13.0.1
Fetching concurrent-ruby 1.1.7
Errno::EACCES: Permission denied @ rb_file_s_rename - (/home/mastodon/.gem/ruby/2.7.0/cache/concurrent-ruby-1.1.7.gem,
/home/mastodon/live/vendor/bundle/ruby/2.7.0/cache/concurrent-ruby-1.1.7.gem)
An error occurred while installing concurrent-ruby (1.1.7), and Bundler cannot continue.
Make sure that `gem install concurrent-ruby -v '1.1.7' --source 'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  health_check was resolved to 4.0.0.pre, which depends on
    rails was resolved to 5.2.4.4, which depends on
      actioncable was resolved to 5.2.4.4, which depends on
        actionpack was resolved to 5.2.4.4, which depends on
          actionview was resolved to 5.2.4.4, which depends on
            rails-dom-testing was resolved to 2.0.3, which depends on
              activesupport was resolved to 5.2.4.4, which depends on
                i18n was resolved to 1.8.5, which depends on
                  concurrent-ruby
```

I checked for permissions on that file and I have rwrwr for mastodon:mastodon, so should be fine.

I tried already to gem install as suggested, the install seems to be completed fine but the bundle install fails again.

I have no luck with v3.3.0 :)

Thanks for your support.

**Comment by jakesullyneytiri** [comment]
> I managed to restore the machine from a snapshot and I started from scratch from v3.2.2, then I did the following:
> 
> 1. su - mastodon
> 2. git -C /home/mastodon/.rbenv/plugins/ruby-build pull
> 3.  rbenv install 2.7.2
> 4. stop mastodon services
> 5. rbenv uninstall 2.6.6
> 6. git fetch --tags
> 7. git checkout v3.3.0 
> 8. bundle install
> 
> I got stuck in here with:
> 
> ```
> mastodon@mastodon:~/live$ bundle install
> Fetching gem metadata from https://rubygems.org/............
> Using rake 13.0.1
> Fetching concurrent-ruby 1.1.7
> Errno::EACCES: Permission denied @ rb_file_s_rename - (/home/mastodon/.gem/ruby/2.7.0/cache/concurrent-ruby-1.1.7.gem,
> /home/mastodon/live/vendor/bundle/ruby/2.7.0/cache/concurrent-ruby-1.1.7.gem)
> An error occurred while installing concurrent-ruby (1.1.7), and Bundler cannot continue.
> Make sure that `gem install concurrent-ruby -v '1.1.7' --source 'https://rubygems.org/'` succeeds before bundling.
> 
> In Gemfile:
>   health_check was resolved to 4.0.0.pre, which depends on
>     rails was resolved to 5.2.4.4, which depends on
>       actioncable was resolved to 5.2.4.4, which depends on
>         actionpack was resolved to 5.2.4.4, which depends on
>           actionview was resolved to 5.2.4.4, which depends on
>             rails-dom-testing was resolved to 2.0.3, which depends on
>               activesupport was resolved to 5.2.4.4, which depends on
>                 i18n was resolved to 1.8.5, which depends on
>                   concurrent-ruby
> ```
> 
> I checked for permissions on that file and I have rwrwr for mastodon:mastodon, so should be fine.
> 
> I tried already to gem install as suggested, the install seems to be completed fine but the bundle install fails again.
> 
> I have no luck with v3.3.0 :)
> 
> Thanks for your support.

Again when checking your new log it's trying to use Ruby 2.7.0 and not 2.7.2 make sure Ruby 2.7.0 is not installed.

**Comment by pixel8383** [comment]
I can't see it:

```
mastodon@mastodon:~/.rbenv/versions$ ls -al
total 12
drwxrwxr-x  3 mastodon mastodon 4096 Jan  2 22:09 .
drwxrwxr-x 13 mastodon mastodon 4096 May 29  2020 ..
drwxrwxr-x  6 mastodon mastodon 4096 Jan  2 22:06 2.7.2
```
As I wrote before, I tried to update Ruby as you suggested.

**Comment by jakesullyneytiri** [comment]
> I can't see it:
> 
> ```
> mastodon@mastodon:~/.rbenv/versions$ ls -al
> total 12
> drwxrwxr-x  3 mastodon mastodon 4096 Jan  2 22:09 .
> drwxrwxr-x 13 mastodon mastodon 4096 May 29  2020 ..
> drwxrwxr-x  6 mastodon mastodon 4096 Jan  2 22:06 2.7.2
> ```
> As I wrote before, I tried to update Ruby as you suggested.

But your error log says different 

```
concurrent-ruby 1.1.7 Errno::EACCES: Permission denied @ rb_file_s_rename - (/home/mastodon/.gem/ruby/2.7.0/cache/concurrent-ruby-1.1.7.gem
```

You see there it says ruby/2.7.0 not 2.7.2

**Comment by pixel8383** [comment]
Uhm, I got it, 2.7.2's cache is located under `/home/mastodon/.gem/ruby/2.7.0/cache` (don't ask me why). I changed the owner recursively on that folder and bundle install completed.

In the end I got:

```
HEADS UP! i18n 1.1 changed fallbacks to exclude default locale.
But that may break your application.

If you are upgrading your Rails application from an older version of Rails:

Please check your Rails app for 'config.i18n.fallbacks = true'.
If you're using I18n (>= 1.1.0) and Rails (< 5.2.2), this should be
'config.i18n.fallbacks = [I18n.default_locale]'.
If not, fallbacks will be broken in your app by I18n 1.1.x.

If you are starting a NEW Rails application, you can ignore this notice.

For more info see:
https://github.com/svenfuchs/i18n/releases/tag/v1.1.0

```
Can I ignore it or should I go on it?

**Comment by pixel8383** [comment]
I am stuck anyway one step later:
I tried to execute SKIP_POST_DEPLOYMENT_MIGRATIONS=true RAILS_ENV=production bundle exec rails db:migrate but it fails with the error you will see in the end of the log. So I tried to run "bundle install -redownload". Everything seems to be installed fine until I get:
```

Bundler::GemNotFound: Could not find openssl-2.2.0.gem for installation
An error occurred while installing openssl (2.2.0), and Bundler cannot continue.
Make sure that `gem install openssl -v '2.2.0' --source 'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  webauthn was resolved to 3.0.0.alpha1, which depends on
    openssl
mastodon@mastodon:~/live$
mastodon@mastodon:~/live$ gem install openssl -v '2.2.0' --source 'https://rubygems.org/'
Building native extensions. This could take a while...
Successfully installed openssl-2.2.0
Parsing documentation for openssl-2.2.0
Done installing documentation for openssl after 1 seconds
1 gem installed
mastodon@mastodon:~/live$ SKIP_POST_DEPLOYMENT_MIGRATIONS=true RAILS_ENV=production bundle exec rails db:migrate
Could not find openssl-2.2.0 in any of the sources
Run `bundle install` to install missing gems.
mastodon@mastodon:~/live$
```

So I am suggested to run once again bundle install.

**Comment by pixel8383** [comment]
I think I fixed it. I had to run `bundle update && bundle install` before running `SKIP_POST_DEPLOYMENT_MIGRATIONS=true RAILS_ENV=production bundle exec rails db:migrate`. After that I was able to follow the instructions. Thanks for your support.
